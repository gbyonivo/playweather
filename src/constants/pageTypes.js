export default {
  forecast: 'forecast',
  current: 'current'
};