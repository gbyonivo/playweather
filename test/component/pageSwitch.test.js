import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import PageSwitch from '../../src/components/pageSwitch';
import combinedReducer from '../../src/reducers';
import pageTypes from '../../src/constants/pageTypes';

const props = {
  fetchData: () => { },
  text: 'scorpion',
  linkTo: 'scorpion king'
};


describe('PageSwitch component without test', () => {
  it('renders correctly', () => {
    const store = createStore(combinedReducer(), { weatherReducer: { activeLocation: 'machetser, uk' } });
    const actual = renderer.create(
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
        <Provider store={store}>
          <PageSwitch {...props} />
        </Provider>
      </MemoryRouter>
    );
    expect(actual).toMatchSnapshot();
  });
});

describe('PageSwitch component without test', () => {
  it('renders correctly', () => {
    const store = createStore(combinedReducer(), { weatherReducer: { activeLocation: 'machetser, uk' } });
    const actual = renderer.create(
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey', match: { params: { type: pageTypes.current } } }]}>
        <Provider store={store}>
          <PageSwitch {...props} />
        </Provider>
      </MemoryRouter>
    );
    expect(actual).toMatchSnapshot();
  });
});