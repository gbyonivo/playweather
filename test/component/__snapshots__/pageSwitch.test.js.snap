// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSwitch component without test renders correctly 1`] = `
<a
  className="pageSwitch"
  href="/forecast"
  onClick={[Function]}
>
  <div
    onClick={[Function]}
  >
    See Current Weather
  </div>
</a>
`;

exports[`PageSwitch component without test renders correctly 2`] = `
<a
  className="pageSwitch"
  href="/forecast"
  onClick={[Function]}
>
  <div
    onClick={[Function]}
  >
    See Current Weather
  </div>
</a>
`;
